src/inverse/SIA2D/VJPs.jl

```diff
@@ -67,3 +68,41 @@ function VJP_λ_∂surface_V∂θ(VJPMode::DiscreteVJP, λx, λy, H, θ, simulat
     λ_∂V∂H = VJP_λ_∂surface_V∂θ_discrete(λx, λy, H, θ, simulation, t)
     return λ_∂V∂H, nothing
 end
+
+
+function MB_wrapper!(MB, H, simulation, glacier)
+    model = simulation.model
+    cache = simulation.cache
+    glacier.S .= glacier.B .+ H
+
+    # Below we call the functions inside MB_timestep! manually
+    # This is because get_cumulative_climate! cannot be differentiated with Enzyme, so it is called beforehand in the VJP function to retrieve the cumulative climate
+    downscale_2D_climate!(glacier)
+    cache.iceflow.MB .= compute_MB(model.mass_balance, glacier.climate.climate_2D_step)
+
+    apply_MB_mask!(H, cache.iceflow)
+    MB .= simulation.cache.iceflow.MB
+end
+function VJP_λ_∂MB∂H(VJPMode::EnzymeVJP, λ, H, simulation::Simulation, glacier, t)
+    # Differentiation of get_cumulative_climate! with Enzyme yields an error
+    # Since it isn't involved in the gradient computation (doesn't depend on H), it can be computed beforehand
+    get_cumulative_climate!(glacier.climate, t, simulation.parameters.solver.step)
+
+    _simulation = Enzyme.make_zero(simulation)
+    _glacier = Enzyme.make_zero(glacier)
+    λ_∂MB∂H = Enzyme.make_zero(H)
+    MB = Enzyme.make_zero(H)
+    λH = deepcopy(λ) # Need to copy because Enzyme changes the backward gradient in-place
+    Enzyme.autodiff(
+        Reverse, MB_wrapper!, Const,
+        Duplicated(MB, λH),
+        Duplicated(H, λ_∂MB∂H),
+        Duplicated(simulation, _simulation),
+        Duplicated(glacier, _glacier),
+    )
```

Comment:
I think is great this is super generic and it should work for any MB model, but notice that the adjoint contribution of the mass balance is actually very simple to compute and it is given by $\frac{\partial MB}{\partial H} \lambda$. For very simple MBs (like the ones we are using), the partial derivative $frac{\partial MB}{\partial H}$ is trivial to compute. This can be the default for some simple MBs models, but I imagine we can have at least one MB where we do this and we just use it for testing. Does this makes sense? Maybe it is a bit too much, but the actual contribution of MB to the adjoint is mathematically very simple, reason why I think we should consider having this done more "manually". What do you think @albangossard ?